### PR TITLE
Implemented solution for longest substring with K unique characters

### PR DESCRIPTION
**Pull request by jpai-15**

### CODE REVIEW

File: solution.py [hunk 1]
```diff
@@ -1,3 +1,71 @@
...
+            #move left point one step to right
+            left += 1
+
+        # Check if the current window has exactly K unique characters
+        if len(char_count) == k:
+            max_len = max(max_len, right - left + 1)
+
+    return max_len if max_len > 0 else 0
+
+
```
Comment (haleynpatel) [inline solution.py:35]: Your test cases are nice and clear! You take into account edge cases well.

File: solution.py [hunk 2]
```diff
@@ -1,3 +1,71 @@
 def longest_substring_k_unique(s: str, k: int) -> int:
-    # Write your code here
-    pass
+#Jhanvi Pai - CS490
+#best case - there is no unique characters to be found
+    if k == 0:
+        return 0
+    
+    # Dictionary to store the count of unique characters in the current window
```
Comment (haleynpatel) [inline solution.py:7]: Commenting what you are doing is a good method.

File: solution.py [hunk 3]
```diff
@@ -1,3 +1,71 @@
 def longest_substring_k_unique(s: str, k: int) -> int:
-    # Write your code here
-    pass
+#Jhanvi Pai - CS490
+#best case - there is no unique characters to be found
+    if k == 0:
```
Comment (haleynpatel) [inline solution.py:4]: Can add more edge cases such as the one mentioned in the prompt where k should be equal to or less than the number of distinct elements in the input.
